alibi_detect/cd/keops/tests/test_mmd_keops.py

```diff
@@ -0,0 +1,100 @@
+from functools import partial
+from itertools import product
+import numpy as np
+import pytest
+import torch
+import torch.nn as nn
+from typing import Callable, List
+from alibi_detect.utils.frameworks import has_keops
+from alibi_detect.cd.pytorch.preprocess import HiddenOutput, preprocess_drift
+if has_keops:
+    from alibi_detect.cd.keops.mmd import MMDDriftKeops
+
+n, n_hidden, n_classes = 500, 10, 5
+
+
+class MyModel(nn.Module):
+    def __init__(self, n_features: int):
+        super().__init__()
+        self.dense1 = nn.Linear(n_features, 20)
+        self.dense2 = nn.Linear(20, 2)
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = nn.ReLU()(self.dense1(x))
+        return self.dense2(x)
+
+
+# test List[Any] inputs to the detector
+def preprocess_list(x: List[np.ndarray]) -> np.ndarray:
+    return np.concatenate(x, axis=0)
+
+
+n_features = [10]
+n_enc = [None, 3]
+preprocess = [
+    (None, None),
+    (preprocess_drift, {'model': HiddenOutput, 'layer': -1}),
+    (preprocess_list, None)
+]
+preprocess_at_init = [True, False]
+n_permutations = [10]
+batch_size_permutations = [10, 1000000]
+configure_kernel_from_x_ref = [True, False]
+tests_mmddrift = list(product(n_features, n_enc, preprocess, n_permutations, preprocess_at_init,
+                              batch_size_permutations, configure_kernel_from_x_ref))
+n_tests = len(tests_mmddrift)
```

Comment:
Noting that `update_x_ref = [{'last': 750}, {'reservoir_sampling': 750}, None]` is missing from the parameters compared to the `torch` and `tensorflow` tests.

Reply:
Done